tools/monorepo-utils/src/code-freeze/commands/version-placeholder/index.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { Command } from '@commander-js/extra-typings';
+import simpleGit from 'simple-git';
+
+/**
+ * Internal dependencies
+ */
+import { Logger } from '../../../core/logger';
+import { cloneRepo, generateDiff, getPatches, getFilename, getPatchAdditions } from '../../../core/git';
+import { getEnvVar } from '../../../core/environment';
+import { validateArgs } from './lib/validate';
+import { updateFileLine } from './lib/update';
+import { Options } from './types';
+
+const genericErrorFunction = ( err ) => {
+	if ( err.git ) {
+		return err.git;
+	}
+	throw err;
+};
+
+export const versionPlaceholderCommand = new Command( 'version-placeholder' )
+	.description( 'Replace version placeholders in DocBlocks' )
+	.requiredOption( '-v, --version <version>', 'Version to replace placeholder with' )
+	.requiredOption(
+		'-p --previous <previous>',
+		'Branch to compare against base for replacement of placeholders'
+	)
+	.option(
+		'-o --owner <owner>',
+		'Repository owner. Default: woocommerce',
+		'woocommerce'
+	)
+	.option(
+		'-n --name <name>',
+		'Repository name. Default: woocommerce',
+		'woocommerce'
+	)
+	.option(
+		'-b --base <base>',
+		'Base branch to create the PR against. Default: trunk',
+		'trunk'
+	)
+	.action( async ( options: Options ) => {
+		const { owner, name, version, base, previous } = options;
+		Logger.startTask(
+			`Making a temporary clone of '${ owner }/${ name }'`
+		);
+		const source = `github.com/${ owner }/${ name }`;
+		const token = getEnvVar( 'GITHUB_TOKEN', true );
+		const remote = `https://${ owner }:${ token }@${ source }`;
+		const tmpRepoPath = await cloneRepo( remote );
+
+		Logger.endTask();
+
+		Logger.notice(
+			`Temporary clone of '${ owner }/${ name }' created at ${ tmpRepoPath }`
+		);
+
+		await validateArgs( version );
+
+		Logger.startTask( `Comparing ${ previous } with ${ base }...` );
+		const diff = await generateDiff( tmpRepoPath, previous, base, () => {
+			throw new Error( `Unable to generate diff betweeen ${ previous } and ${ base }` );
```

Comment:
This error function swallows the error generated by `generateDiff`. I think it would be more helpful to `Logger.error` the argument passed.

In my case the error is `Error: error: pathspec 'test/placeholder-replacement' did not match any file(s) known to git`.

I don't think the branch `test/placeholder-replacement` is ever pulled down from the remote at this point? We have a helper function called `checkoutRemoteBranch` which may be helpful.